servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancedStreamingHttpClient.java

```diff
@@ -121,8 +121,10 @@ public void cancel() {
     @Override
     public Single<ReservedStreamingHttpConnection> reserveConnection(final HttpExecutionStrategy strategy,
                                                                      final HttpRequestMetaData metaData) {
-        return strategy.offloadReceive(executionContext.executor(),
-                loadBalancer.selectConnection(SELECTOR_FOR_RESERVE).map(identity()));
+        Single<ReservedStreamingHttpConnection> connection =
+                loadBalancer.selectConnection(SELECTOR_FOR_RESERVE).map(identity());
+        return strategy.isMetadataReceiveOffloaded() || strategy.isDataReceiveOffloaded() ?
```

Comment:
future: `reserveConnection` doesn't actually make a request, but just marks it as ineligible for the LB to let anyone else select the connection. from the `HttpExecutionStrategy` this would likely fall on the control path and the only way we will be on the event loop is if the `ConnectionFactory` created a new connection and is calling us back from the EventLoop.